haskell-tests/Types/CBOR.hs

```diff
@@ -295,17 +313,101 @@ testEncodedTokenOperations = describe "EncodedTokenOperations JSON serialization
+                "\xA2\x63\x75\x72\x6C\x76\x68\x74\x74\x70\x73\x3A\x2F\x2F\x61\x62\x63\x2E\x74\x6F\x6B\x65\x6E\x2F\x6D\x65\x74\x61\x6E\x63\x68\x65\x63\x6B\x73\x75\x6D\x53\x68\x61\x32\x35\x36\x58\x20\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB"
+            )
+    it "Decode TokenMetadataUrl with only url" $
+        assertEqual
+            "Decoded TokenMetadataUrl"
+            ( Right
+                ( "",
+                  TokenMetadataUrl
+                    { tmUrl = "https://abc.token/meta",
+                      tmChecksumSha256 = Nothing,
+                      tmAdditional = Map.empty
+                    }
+                )
+            )
+            ( deserialiseFromBytes
+                decodeTokenMetadataUrl
+                "\xA1\x63\x75\x72\x6C\x76\x68\x74\x74\x70\x73\x3A\x2F\x2F\x61\x62\x63\x2E\x74\x6F\x6B\x65\x6E\x2F\x6D\x65\x74\x61"
+            )
+    it "Decode TokenMetadataUrl with url, checksumSha256, and additional fields" $
+        assertEqual
+            "Decoded TokenMetadataUrl"
+            ( Right
+                ( "",
+                  TokenMetadataUrl
+                    { tmUrl = "https://abc.token/meta",
+                      tmChecksumSha256 = Just $ SHA256.Hash (FBS.pack $ replicate 32 0xab),
+                      tmAdditional =
+                        Map.fromList
+                            [ ("key1", CBOR.TInt 42),
+                              ("key2", CBOR.TString "extra value")
+                            ]
+                    }
+                )
+            )
+            ( deserialiseFromBytes
+                decodeTokenMetadataUrl
+                "\xA4\x63\x75\x72\x6C\x76\x68\x74\x74\x70\x73\x3A\x2F\x2F\x61\x62\x63\x2E\x74\x6F\x6B\x65\x6E\x2F\x6D\x65\x74\x61\x6E\x63\x68\x65\x63\x6B\x73\x75\x6D\x53\x68\x61\x32\x35\x36\x58\x20\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\xAB\x64\x6B\x65\x79\x31\x18\x2A\x64\x6B\x65\x79\x32\x6B\x65\x78\x74\x72\x61\x20\x76\x61\x6C\x75\x65"
+            )
+
 tests :: Spec
 tests = parallel $ describe "CBOR" $ do
     testInitializationParameters
     testEncodedInitializationParameters
     testEncodedTokenOperations
+    testTokenMetadataUrlJSON
+    testTokenMetadataUrlCBOR
     it "Encode and decode TokenTransfer" $ withMaxSuccess 1000 $ forAll genTokenTransfer $ \tt ->
         (Right ("", tt))
             === ( deserialiseFromBytes
                     decodeTokenTransfer
                     (toLazyByteString $ encodeTokenTransfer tt)
                 )
+    it "Encode and decode TokenMetadataUrl" $ withMaxSuccess 1000 $ forAll genTokenMetadataUrl $ \tt ->
+        (Right ("", tt))
+            === ( deserialiseFromBytes
+                    decodeTokenMetadataUrl
+                    (toLazyByteString $ encodeTokenMetadataUrl tt)
+                )
```

Comment:
I recommend also doing this with JSON encoding, and having tests with/without additional fields. (Could make the generator 50/50 on whether additional fields are included, rather than having two separate generators.)